cloudvolume/cloudvolume.py

```diff
@@ -238,31 +242,19 @@ def refresh_provenance(self):
     return self
 
   def commit_provenance(self):
-    if self._protocol == 'boss':
+    if self.path.protocol == 'boss':
       return self
 
     self._storage.put_file('provenance', self.provenance.serialize(), content_type='application/json')
     return self
 
   @property
   def dataset_name(self):
-    return self._dataset_name
-
-  @dataset_name.setter
-  def dataset_name(self, name):
-    if name != self._dataset_name:
-      self._dataset_name = name
-      self.refresh_info()
+    return self.path.dataset
```

Comment:
How much time are we saving the user on this?
he can just use the path to get to dataset himself.